master/pkg/schemas/copy.go

```diff
@@ -5,12 +5,30 @@ import (
 	"reflect"
 )
 
+// Copyable means an object can have custom behvaiors for schemas.Copy.  Mostly this is useful for
+// working around types which we do not own and which schemas.Copy() would puke on.
+type Copyable interface {
+	// Copy should return the same type.  It must not be defined as a method against a pointer of
+	// the type or it will not work.
+	Copy() interface{}
+}
+
 // cpy is for deep copying, but it will only work on "nice" objects, which should include our
 // schema objects.  Useful to other reflect code.
 func cpy(v reflect.Value) reflect.Value {
 	// fmt.Printf("cpy(%T)\n", v.Interface())
 	var out reflect.Value
 
+	// Detect Copyables, but disallow pointers.  The reason is that if you have a method like:
+	//    func (t Thing) Copy() interface{}
+	// then Copy() will return a plain Thing object, but a pointer to a Thing will still be treated
+	// as copyable.  Then, schemas.Copy(&t).(*Thing) would panic because it returns the wrong type.
```

Comment:
Seems to me a method `func (t Thing) Copy() interface{}` should always operate on non-pointer receivers not the opposite?

Reply:
`obj.Kind() != reflect.Ptr` == brad cant read

Reply:
Is that not what I'm saying?  Suppose you have a
```go
func (t Thing) Copy() interface{}
```
then you have to be careful as you are recursing through something like:
```go
type SuperThing struct {
    RawThing *Thing
}
```
then you'll find that:
```
_, ok =: superThing.RawThing.(Copyable)
```
Will return `ok`.  But that's a problem!  Because you will have accidentally converted a pointer to a non-pointer as part of `schemas.Copy()`.